### ISSUE TITLE
5-fold cross validation in DeepEthogram

### ISSUE THREAD

**Issue by monajalal** [jbohnslav/deepethogram#25]
Hi Jim,

If we want to do 5-fold cross validation, should we manually have the following (an example):

1. One 8 min video (train)
2. One 1 min video (test)
3. One 1 min video (validation)

I am not sure how to randomly create these random 8, 1, 1 continuous minute selections of a 10 video for 5-fold CV in order to go about annotating it using DeepEthogram?

If it was photo, I would use scikit-learn. But I am not aware how to do it for video. Any tip, suggestion, or link to external resources is highly appreciated.

**Comment by jbohnslav** [comment]
Hmm, this is an interesting question. The GUI / project structure in DeepEthogram is pretty much designed for static individual videos, rather than splits within a certain video. If you look at static temporal action detection datasets, e.g. [Thumos14](http://crcv.ucf.edu/THUMOS14/download.html) or [AVA](https://arxiv.org/abs/1705.08421), the splits are all done at the video level-- so there isn't an established example to go on. 

I think the easiest way is probably to split your 10-minute video into 1 minute chunks and add them to a deepethogram project (e.g. with `projects.add_video_to_project`). You can easily split up your labels, which are just arrays. Then you can generate your own train / validation / test splits and save each of them to a YAML file with this structure:

```yaml
metadata: # note: I'm pretty sure metadata isn't used except to add additional videos programatically
  split:
  - 0.8
  - 0.1
  - 0.1
test: []
train:
- clip00
- clip01
# ...
val: 
- clip02
- clip03
# ...
test: 
- clip04
- clip05
# ...
```

Then you can pass `split.file=path/to/split1.yaml` as a command line argument when training. This is probably the easiest way with the current codebase. 

You can, of course, use the provided architectures and training scripts, and make your own dataloader that can take in individual chunks of videos being in individual splits rather than the whole thing.
